.github/workflows/ci.yml

```diff
@@ -0,0 +1,114 @@
+# ci.yml
+#
+# Continuous Integration for Chemprop - checks build, code formatting, and 
+# runs tests for all proposed changes and on a regular schedule
+#
+# Note: this file contains extensive inline documentation to aid with
+# knowledge transfer.
+
+name: Continuous Integration
+
+on:
+  # run on pushes/pull requests to/against main or v2/dev
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main, v2/dev]
+  # run this in the morning on weekdays to catch dependency issues
+  schedule:
+    - cron: "0 8 * * 1-5"
+  # allow manual runs
+  workflow_dispatch:
+
+# cancel previously running tests if new commits are made
+# https://docs.github.com/en/actions/examples/using-concurrency-expressions-and-a-test-matrix
+concurrency:
+  group: actions-id-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Check Build
+    runs-on: ubuntu-latest
+    steps:
+      # clone the repo, attempt to build
+      - uses: actions/checkout@v4
+      - run: python -m pip install build
+      - run: python -m build .
+
+  lint:
+    name: Check Formatting
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      # clone the repo, run black and autopep8 on it
```

Comment:
Is this comment right? We are running flake8 not autopep8?

Reply:
good catch, we are running `flake8` which reports the errors that would be fixed by `autopep8`. I will fix this after the CI runs